Applications/mnist/jni/main.cpp

```diff
@@ -0,0 +1,293 @@
+
+/**
+ * Copyright (C) 2020 Jijoong Moon <jijoong.moon@samsung.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0-only
+ *
+ * @file	main.cpp
+ * @date	01 July 2020
+ * @see		https://github.com/nnstreamer/nntrainer
+ * @author	Jijoong Moon <jijoong.moon@samsung.com>
+ * @bug		No known bugs except for NYI items
+ * @brief	This is MNIST Example with
+ * 
+ * Input 28x28
+ * Conv2D 5 x 5 : 6 filters, stride 1x1, padding=0,0
+ * Pooling2D : 2 x 2, Average pooling, stride 1x1
+ * Conv2D 6 x 5 x 5 : 12 filters, stride 1x1, padding=0,0
+ * Pooling2D : 2 x 2, Average Pooling, stride 1x1
+ * Flatten
+ * Fully Connected Layer with 10 unit
+ *
+ */
+
+#include <climits>
+#include <cmath>
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <queue>
+#include <sstream>
+#include <stdlib.h>
+#include <time.h>
+
+#include "neuralnet.h"
+#include "tensor.h"
+
+#define TRAINING true
+
+/**
+ * @brief     Data size for each category
+ */
+const unsigned int total_train_data_size = 100;
+
+const unsigned int total_val_data_size = 100;
+
+const unsigned int total_test_data_size = 100;
+
+const unsigned int buffer_size = 100;
+
+/**
+ * @brief     Number of category : Three
+ */
+const unsigned int total_label_size = 10;
+
+/**
+ * @brief     Max Epoch
+ */
+const unsigned int iteration = 300;
+
+const unsigned int mini_batch = 32;
+
+const unsigned int feature_size = 784;
+
+std::string data_path;
+
+bool duplicate[total_label_size * total_train_data_size];
+bool valduplicate[total_label_size * total_val_data_size];
+
+
+/**
+ * @brief     step function
+ * @param[in] x value to be distinguished
+ * @retval 0.0 or 1.0
+ */
+float stepFunction(float x) {
+  if (x > 0.9) {
```

Comment:
How about add a tolerance variable and then compare `x + tolerance > 1.0` and `x - tolerance < 0.0`?

Reply:
You may aware that this is not used for comparison. we can do this with tolerance though.